src/main/java/mat/dao/impl/FeatureFlagDAOImpl.java

```diff
@@ -24,16 +20,8 @@ public FeatureFlagDAOImpl(@Autowired SessionFactory sessionFactory) {
     }
 
     @Override
-    public FeatureFlag findFlagByName(String flagName) {
-        final Session session = getSessionFactory().getCurrentSession();
-        final CriteriaBuilder cb = session.getCriteriaBuilder();
-        final CriteriaQuery<FeatureFlag> query = cb.createQuery(FeatureFlag.class);
-        final Root<FeatureFlag> root = query.from(FeatureFlag.class);
-
-        query.select(root).where(cb.equal(root.get(FLAG_NAME), flagName));
-
-        final List<FeatureFlag> dataTypeList = session.createQuery(query).getResultList();
-
-        return CollectionUtils.isNotEmpty(dataTypeList) ? dataTypeList.get(0) : null;
+    public List<FeatureFlag> findAllFeatureFlags() {
+        final List<FeatureFlag> dataTypeList = find();
```

Comment:
There is a well-known idiom: "Prefer empty collections to nulls"
http://www.javapractices.com/topic/TopicAction.do?Id=59
Should we follow it in our project?

Reply:
I suggest that Dao implementation should retrieve a list of FeatureFlags and then transform into a map.

Reply:
Updated